### PR TITLE
lefthook: 0.7.2 -> 0.7.3

### PR DESCRIPTION
**Pull request by NobbZ**
###### Motivation for this change

Lefthook (finally) got updated upstream. This fixes the problem that it can't be
used when built with a Go newer than 1.14 (which is marked insecure).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5950,9 +5950,7 @@ in
   lbreakout2 = callPackage ../games/lbreakout2 { };
 
   lefthook = callPackage ../applications/version-management/git-and-tools/lefthook {
-    # Please use empty attrset once upstream bugs have been fixed
-    # https://github.com/Arkweid/lefthook/issues/151
-    buildGoModule = buildGo114Module;
+    buildGoModule = buildGoModule;
```
Comment (lourkeur) [inline pkgs/top-level/all-packages.nix:5953]:
```suggestion
```
That ought to be redundant

**Comment by lourkeur** [comment]
Result of `nixpkgs-review pr 120346` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lefthook</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -5950,9 +5950,6 @@ in
   lbreakout2 = callPackage ../games/lbreakout2 { };
 
   lefthook = callPackage ../applications/version-management/git-and-tools/lefthook {
-    # Please use empty attrset once upstream bugs have been fixed
-    # https://github.com/Arkweid/lefthook/issues/151
-    buildGoModule = buildGo114Module;
   };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:5953]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -5950,9 +5950,6 @@ in
   lbreakout2 = callPackage ../games/lbreakout2 { };
 
   lefthook = callPackage ../applications/version-management/git-and-tools/lefthook {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:5952]:
```suggestion
  lefthook = callPackage ../applications/version-management/git-and-tools/lefthook { };
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (NobbZ) [reply pkgs/top-level/all-packages.nix]: Indeed :D This happens if one rushes through getting the PR ready on the train via SSH on a mobile :D

**Pull request merged**